Zoom/ZoomSettings.ascx.cs

```diff
@@ -0,0 +1,207 @@
+            _apiSecret = Settings.GetApiSecret();
+            _webhookURL = string.Format( "{0}Plugins/rocks_kfs/Zoom/Webhook.ashx", GlobalAttributesCache.Get().GetValue( "InternalApplicationRoot" ) );
+
+            if ( !Page.IsPostBack )
+            {
+                ShowDetail();
+            }
+        }
+
+        #endregion Control Methods
+
+        #region Events
+
+        /// <summary>
+        /// Handles the BlockUpdated event of the control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void Block_BlockUpdated( object sender, EventArgs e )
+        {
+            ShowDetail();
+        }
+
+        /// <summary>
+        /// Handles the Click event of the btnEdit control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void btnEdit_Click( object sender, EventArgs e )
+        {
+            nbNotification.Visible = false;
+            pnlApiSettings.Visible = true;
+            HideSecondaryBlocks( true );
+            lView.Visible = false;
+            btnEdit.Visible = false;
+            btnSyncNow.Visible = false;
+        }
+
+        /// <summary>
+        /// Handles the Click event of the btnSave control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void btnSave_Click( object sender, EventArgs e )
+        {
+            Settings.SaveApiSettings( tbApiKey.Text, tbApiSecret.Text );
+            _apiKey = tbApiKey.Text;
+            _apiSecret = tbApiSecret.Text;
+            nbNotification.Text = "API Settings Saved!";
+
+            pnlApiSettings.Visible = false;
+            HideSecondaryBlocks( false );
+            nbNotification.Visible = true;
+            nbNotification.Title = "";
+            nbNotification.NotificationBoxType = Rock.Web.UI.Controls.NotificationBoxType.Info;
+            nbNotification.Dismissable = true;
+            ShowDetail( true );
+        }
+
+        /// <summary>
+        /// Handles the Click event of the btnCancel control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void btnCancel_Click( object sender, EventArgs e )
+        {
+            pnlApiSettings.Visible = false;
+            HideSecondaryBlocks( false );
+            nbNotification.Visible = false;
+            ShowDetail();
+        }
+
+        protected void btnSyncNow_Click( object sender, EventArgs e )
+        {
+            using ( var rockContext = new RockContext() )
```

Comment:
Since context is not being used anywhere else, why pass it in? Move context into method